### PR TITLE
Zarch fix gcc detection

### PR DESCRIPTION
**Pull request by ghost**
Fix the build failure with gcc-10 and DYNAMIC_ARCH=1 on s390x. Specifically, fix and tidy up the check for gcc version >= 5.2 by employing and extending the common version check variables, as introduced in #2626

### CODE REVIEW

File: Makefile.system [hunk 1]
```diff
@@ -606,7 +606,6 @@ ifneq ($(C_COMPILER), GCC)
 DYNAMIC_CORE += POWER9
 endif
 ifeq ($(C_COMPILER), GCC)
-GCCVERSIONGT5 := $(shell expr `$(CC) -dumpversion | cut -f1 -d.` \> 5)
```
Comment (RajalakshmiSR) [inline Makefile.system:600]: @mhillenibm This is needed for POWER as the variable was set under `ifeq ($(OSNAME), WINNT)` in the previous lines..

Comment (RajalakshmiSR) [reply Makefile.system:600]: Ok.. they are not arch specific now.. This is fine then .

Comment (martin-frbg) [reply Makefile.system:600]: it was actually rescued out of the WINNT-only section by #2626 so de-duplicating should be ok

Comment (ghost) [reply Makefile.system:600]: note that the definition has moved out of the OS-specific section, in #2626

**Pull request merged**